code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -122,8 +122,8 @@
 	atoms_to_ravage += get_step(owner, turn(owner.dir, -45)).contents
 	atoms_to_ravage += get_step(owner, turn(owner.dir, 45)).contents
 	for(var/atom/movable/ravaged AS in atoms_to_ravage)
-		if(ishitbox(ravaged) || isarmoredvehicle(ravaged))
-			ravaged.attack_alien(X, X.xeno_caste.melee_damage) //Handles APC/Tank stuff
+		if(ishitbox(ravaged) || isvehicle(ravaged))
+			ravaged.attack_alien(X, X.xeno_caste.melee_damage) //Handles APC/Tank stuff. Has to be before the !ishuman check or else ravage does work properly on vehicles.
 			continue
 		if(!(ravaged.resistance_flags & XENO_DAMAGEABLE) || !X.Adjacent(ravaged))
```

Comment:
Note that you've put this before the adjacency check. I assume you've done this because the hitbox probably fails the adjacency check or something?

Either way, it means in some cases you'll be able to hit vehicles through walls which is uh, not entirely desirable.

Reply:
Correct, putting this after the adjacency check causes it to fail unless you are literally on top of the vehicle (or somehow right under it).  I'm unsure how to keep the adjacency check before the hitbox check without it not working.